src/CandidateComparator.php

```diff
@@ -3,10 +3,10 @@
 
 use \InvalidArgumentException;
 
-class CandidateComparator
+abstract class CandidateComparator
```

Comment:
Made this abstract so that subclasses could vary their handling of unknown candidates while sharing other candidate comparison functionality.